cpp/include/raft/sparse/neighbors/detail/connect_components.cuh

```diff
@@ -324,34 +435,45 @@ void min_components_by_color(raft::sparse::COO<value_t, value_idx>& coo,
  * @param[out] out output edge list containing nearest cross-component
  *             edges.
  * @param[in] X original (row-major) dense matrix for which knn graph should be constructed.
- * @param[in] colors array containing component number for each row of X
+ * @param[in] orig_colors array containing component number for each row of X
  * @param[in] n_rows number of rows in X
  * @param[in] n_cols number of cols in X
+ * @param[in] reduction_op reduction operation for computing nearest neighbors. The reduction
+ * operation must have `gather` and `scatter` functions defined. For single linkage clustering,
+ * these functions are no-ops. For HDBSCAN, they sort and 'unsort' the core distances based on color
+ * @param[in] row_batch_size the batch size for computing nearest neighbors. This parameter controls
+ * the number of samples for which the nearest neighbors are computed at once. Therefore, it affects
+ * the memory consumption mainly by reducing the size of the adjacency matrix for masked nearest
+ * neighbors computation. default 0 indicates that no batching is done
+ * @param[in] col_batch_size the input data is sorted and 'unsorted' based on color. An additional
+ * scratch space buffer of shape (n_rows, col_batch_size) is created for this. Usually, this
+ * parameter affects the memory consumption more drastically than the col_batch_size with a marginal
+ * increase in compute time as the col_batch_size is reduced. default 0 indicates that no batching
+ * is done
  */
 template <typename value_idx, typename value_t, typename red_op>
-void connect_components(
-  raft::resources const& handle,
-  raft::sparse::COO<value_t, value_idx>& out,
-  const value_t* X,
-  const value_idx* orig_colors,
-  size_t n_rows,
-  size_t n_cols,
-  red_op reduction_op,
-  raft::distance::DistanceType metric = raft::distance::DistanceType::L2SqrtExpanded)
+void connect_components(raft::resources const& handle,
+                        raft::sparse::COO<value_t, value_idx>& out,
```

Comment:
Just making a note that at some point we need to update this API to use our new `raft::device_coo_matrix` types. Nothing to change here yet.